web-api/tests/recommendations.rs

```diff
@@ -122,311 +84,103 @@ macro_rules! assert_order {
         );
         for documents in $documents.windows(2) {
             let [d1, d2] = documents else { unreachable!() };
-            assert!(d1.score >= d2.score, $($arg)*);
-            assert!(d2.score >= 0., $($arg)*);
+            assert!(1. >= d1.score, $($arg)*);
+            assert!(d1.score > d2.score, $($arg)*);
+            assert!(d2.score >= -1., $($arg)*);
```

Comment:
:wrench:  This is wrong.

We don't give any guarantees about the range scores are in.

Through we currently should only have scores in range [0;1],
but that could change but we kinda don't want to return scores < 0.

(Similarities tend to be in range [-1;1] but we return scores not similarities.)

Reply:
I simply copied the code from the previous recommendations test, that now is called user_recommendations

Reply:
I removed that check

Reply:
The diff says you replaced:

```rust
 assert!(d1.score >= d2.score, $($arg)*);
 assert!(d2.score >= 0., $($arg)*);
```

with

```rust
assert!(1. >= d1.score, $($arg)*);
assert!(d1.score > d2.score, $($arg)*);
assert!(d2.score >= -1, $($arg)*);
```

but the diff seems also to be _very_ confused :thinking: 

I think you might have copied from a outdated pice of code which was already fixed upstream.


```rust
            // While score can be any arbitrary value it _currently_ should happen to be in [0;1]
            assert!(1. >= d1.score, $($arg)*);
            assert!(d1.score >= d2.score, $($arg)*);
            assert!(d2.score >= 0., $($arg)*);
```

(hit: this doesn't just change the bound from -1 to 0 but also the `>` to a `>=` as both scores could very well be the same)